assets/js/modules/analytics/common/account-create/country-select.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * CountrySelect component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	Select,
+	Option,
+} from '../../../../material-components';
+import { allCountries, countriesByCode } from '../../util/countries-timezones';
+import Data from 'googlesitekit-data';
+import { STORE_NAME, FORM_ACCOUNT_CREATE } from '../../datastore/constants';
+const { useSelect, useDispatch } = Data;
+
+export default function CountrySelect() {
+	const value = useSelect( ( select ) => select( STORE_NAME ).getForm( FORM_ACCOUNT_CREATE, 'countryCode' ) );
+
+	const { setForm } = useDispatch( STORE_NAME );
+	const onEnhancedChange = useCallback( ( i, item ) => {
+		const newCountryCode = item.dataset.value;
+		if ( newCountryCode !== value ) {
+			setForm( FORM_ACCOUNT_CREATE, {
+				countryCode: newCountryCode,
+				timezone: countriesByCode[ newCountryCode ].defaultTimeZoneId,
```

Comment:
This is _probably_ not going to happen since the potential values are sourced by the same data, but it may be a bit safer to add a check for `countriesByCode[ newCountryCode ]` in the if clause above. If it were to be `undefined` here, we'd get a JavaScript error.

But again, given the data this should never happen. I'd rather do defensive coding regardless.